### ISSUE TITLE
Possibility to configure for single-indent parentheses?

### ISSUE THREAD

**Issue by lihaoyi** [olafurpg/scalafmt#143]
This is the style I've been using for a while, and i've seen it used in a bunch of places.

Original:
```scala
object FormatMe { List(number) match { case head :: Nil if head % 2 == 0 => "number is even"
  case head :: Nil => "number is not even"
  case Nil => "List is empty" }
  function(arg1, arg2(arg3(arg4, arg5, "arg6"), arg7 + arg8), arg9.select(1, 2, 3, 4, 5, 6)) }
```

Scalafmt:
```scala
object FormatMe {
  List(number) match {
    case head :: Nil
        if head % 2 == 0 =>
      "number is even"
    case head :: Nil =>
      "number is not even"
    case Nil => "List is empty"
  }
  function(
      arg1,
      arg2(arg3(arg4, arg5, "arg6"),
           arg7 + arg8),
      arg9.select(1, 2, 3, 4, 5, 6))
}
```

I would like the input to look like this:
```scala
object FormatMe {
  List(number) match {
    case head :: Nil
        if head % 2 == 0 =>
      "number is even"
    case head :: Nil =>
      "number is not even"
    case Nil => "List is empty"
  }
  function(
    arg1,
    arg2(
      arg3(arg4, arg5, "arg6"),
      arg7 + arg8
    ),
    arg9.select(1, 2, 3, 4, 5, 6)
  )
}
```

Using
* 0.1.x

Main points:

- `(`s result in a two-space indent, just like `{`s
- `(`s and `)`s are new-lined just like `{`s and `}`s

This is most useful when working with non-trivial nested expressions, e.g. Scalatags HTML templates:

```scala
html(
  head(
    script(src:="..."),
    script(
      "alert('Hello World')"
    )
  ),
  body(
    div(
      h1(id:="title", "This is a title"),
      p("This is a big paragraph of text")
    )
  )
)
```
```scala

  html(
    head(
      script("some script")
    ),
    body(
      h1("This is my title"),
      div("posts"),
      for ((name, text) <- posts) yield div(
        h2("Post by ", name),
        p(text)
      ),
      if(numVisitors > 100) p("No more posts!")
      else p("Please post below...")
    )
  )
```

Which result in huge amounts of indentation using double-indents, and look totally unreadable using the current lisp-style parentheses since the large numbers of closing parentheses bunched together messes with your head. OTOH with the proposed formatting style it is laid out almost exactly the same as HTML is laid out and is just as readable

**Comment by olafurpg** [comment]
I agree it should possible to configure the continuation indent for `(`s. I' experienced the same issue you describe with Scalatags HTML templates, it's almost impossible to use 4 space indentation and keep within the 80 column limit.

However, what do you think the indentation should be for arguments of `def`/`class`?

```scala
// column          |
def point(
  x: Int,
  y: Int): Point =
  Point(x, y)
```
It bothers me that the arguments align with the function body. If you agree, what do you think the two configuration options should be named? One suggestion, `continuationIndentCallSite` and `continuationIndentDefinitionSite`.

I just released 0.1.6 ([changelog](http://olafurpg.github.io/scalafmt/#0.1.6)), which makes a lot of progress on formatting almost any arbitrary Scala code. I hope to start working soon on 0.2.x, which will focus on adding configuration options for different coding styles, such as this issue.

**Comment by lihaoyi** [comment]
I've been using

```scala
def point(x: Int,
          y: Int): Point = {
  Point(x, y)
}
```

Why? Because it's IntelliJ's default, and that's a strong enough reason for me. I've seen other people use it too. This also works

```scala

```scala
def point(
  x: Int,
  y: Int
): Point = {
  Point(x, y)
}
```
If you want to be consistent with the above indentation, you just have to be consistent with the placement of parens, and it all looks great. Looks better than the earlier case when the arguments or their types are very long and maybe even need to start wrapping.

I don't do this because IntelliJ does it differently but i don't feel strongly about either.

**Comment by lihaoyi** [comment]
To answer your other question about what to call it, I have no preference, pick whatever you want =)

**Comment by olafurpg** [comment]
[Here](http://localhost:3000/readme/target/scalatex/#Functionapplication) is what scalafmt does at the moment. It will try to align by the opening `(` (option 3), but in the example above that will break the column limit at `|`

```scala
// column          |
def point(x: Int,
          y: Int): Point = {
  Point(x, y)
}
``` 

The second suggestion you propose is the "config" style (option 5), which is opt-in by adding a newline before closing `)`. Option 4 is the same as config style except it doesn't put newline before the closing `)`.

I'll go with both `continuationIndentCallSite` and `continuationIndentDefinitionSite` when I implement this then, in your case you can use 2 for both and opt-in for config style when you want a newline before `)`, indented at 0.

**Comment by olafurpg** [comment]
Just released 0.2.0. You can now create a `.scalafmt` file in the root of your project with

```
--continuationIndentCallSite 2
--continuationIndentDefnSite 2
```

The intellij plugin should pick it up. Unfortunately, you need to restart intellij if you edit `.scalafmt`, I hope to make that easier some future release. More docs on custom configuration are [here](https://olafurpg.github.io/scalafmt/#Configuration).
